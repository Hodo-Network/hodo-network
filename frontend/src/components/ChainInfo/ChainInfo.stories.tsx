import { Meta, Story } from '@storybook/react';

import { PureChainInfo, PureChainInfoProps } from './ChainInfo';

export default {
  title: 'Components/ChainInfo',
  component: PureChainInfo,
} as Meta;

const Template: Story<PureChainInfoProps> = (args) => (
  <PureChainInfo {...args} />
);

export const Default = Template.bind({});
Default.args = {};
