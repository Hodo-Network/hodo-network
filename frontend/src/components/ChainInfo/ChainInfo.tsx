import Button from "../../base/Button";
// import Avalanche from "avalanche";

// let avalanche = new Avalanche("127.0.0.1", 9650, "https");
// const xchain = avalanche.XChain();
// const cchain = avalanche.CChain();
// const cKeychain = cchain.keyChain();
// const cAddressStrings = cchain.keyChain().getAddressStrings();
// const cChainBlockchainID: string = Defaults.network["12345"].C.blockchainID;

export interface ChainInfoProps {
  params: any;
}

export const ChainInfo = ({ params }: ChainInfoProps) => {
  const getBalances = async () => {
    console.log("chaininfo");
    try {
      // @ts-ignore
      await ethereum
        .request({
          method: "eth_getAssetBalance",
          params: [params],
        })
        .then((response: any) => console.log(response));
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <Button color="primary" size='lg' onClick={getBalances}>
      Get Balances
    </Button>
  );
};
