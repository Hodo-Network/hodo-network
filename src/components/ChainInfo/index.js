import React from "react";
import Avalanche from "avalanche";

let avalanche = new Avalanche("127.0.0.1", 9650, "https");
// const xchain = avalanche.XChain();
// const cchain = avalanche.CChain();
// const cKeychain = cchain.keyChain();
// const cAddressStrings = cchain.keyChain().getAddressStrings();
// const cChainBlockchainID: string = Defaults.network["12345"].C.blockchainID;

export default function ChainInfo() {
  const params = {
    address: "0x6cBEd289596f757fC406eE3AB4aC9E83cFb48B92",
    blk: "latest",
    assetID: "3RvKBAmQnfYionFXMfW5P8TDZgZiogKbHjM8cjpu16LKAgF5T",
  };
  const getBalances = async () => {
    try {
      await window.ethereum
        .request({
          method: "eth_getAssetBalance",
          params: [params],
        })
        .then((response) => console.log(response))
        .catch((error) => console.error);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <button className='btn btn--large' onClick={getBalances}>
      Get Balances
    </button>
  );
}
