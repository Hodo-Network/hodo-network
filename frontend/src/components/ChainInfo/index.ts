import { ChainInfo } from './ChainInfo';
export { ChainInfo };
