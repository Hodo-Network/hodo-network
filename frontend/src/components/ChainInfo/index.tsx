import { PureChainInfo } from './ChainInfo';
// import Avalanche from "avalanche";

// let avalanche = new Avalanche("127.0.0.1", 9650, "https");
// const xchain = avalanche.XChain();
// const cchain = avalanche.CChain();
// const cKeychain = cchain.keyChain();
// const cAddressStrings = cchain.keyChain().getAddressStrings();
// const cChainBlockchainID: string = Defaults.network["12345"].C.blockchainID;

// Example
// params: {
//   address: "0x6cBEd289596f757fC406eE3AB4aC9E83cFb48B92",
//   blk: "latest",
//   assetID: "3RvKBAmQnfYionFXMfW5P8TDZgZiogKbHjM8cjpu16LKAgF5T",
// },

export interface ChainInfoProps {
  params: any;
}

const ChainInfo = ({ params }: ChainInfoProps) => {
  const onGetBalances = async () => {
    console.log('chaininfo');
    try {
      // @ts-ignore
      await ethereum
        .request({
          method: 'eth_getAssetBalance',
          params: [params],
        })
        .then((response: any) => console.log(response));
    } catch (error) {
      console.log(error);
    }
  };

  return <PureChainInfo onClick={onGetBalances} />;
};

export default ChainInfo;
