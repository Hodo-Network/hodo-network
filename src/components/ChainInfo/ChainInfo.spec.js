import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ChainInfo from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ChainInfo />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<ChainInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
