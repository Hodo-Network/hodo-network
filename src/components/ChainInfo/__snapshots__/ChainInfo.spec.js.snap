// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="btn btn--large"
  onClick={[Function]}
>
  Get Balances
</button>
`;
